profile-specific-challenges/frontend/components/idioms.jsx

```diff
@@ -145,216 +154,283 @@ export function UntraceableState() {
+        name="age"
+        type="number"
+        onChange={handleFormValueChange}
+      />
+      <input
+        value={location}
+        name="location"
+        type="text"
+        onChange={handleFormValueChange}
+      />
+      <input
+        value={email}
+        name="email"
+        type="email"
+        onChange={handleFormValueChange}
+      />
+      <input
+        value={password}
+        name="password"
+        type="password"
+        onChange={handleFormValueChange}
+      />
+      <button type="submit">Submit</button>
+    </form>
+  );
 }
 
 export function UnidiomaticHTMLHierarchy() {
-  const bids = [1,2,3]
-  const asks = [1,2,3]
+  const bids = [1, 2, 3];
+  const asks = [1, 2, 3];
 
-  return (<li>
-    {bids.map((bid, i) => <span key={i}>{bid}</span>)}
-    {asks.map((ask, j) => <span key={j+'asks'}>{ask}</span>)}
-  </li>)
+  return (
+    <li>
+      {bids.map((bid, i) => (
+        <span key={i}>{bid}</span>
+      ))}
+      {asks.map((ask, j) => (
+        <span key={j + "asks"}>{ask}</span>
+      ))}
+    </li>
+  );
 }
 
 export function SubstandardDataStructure() {
-  const [error, setError] = useState("")
+  const [error, setError] = useState("");
 
-  return(
+  return (
     <div>
-      <button onClick={() => setError('Error A')}>Throw Error A</button>
-      <button onClick={() => setError('Error B')}>Throw Error B</button>
-      <button onClick={() => setError('')}>Clear Errors</button>
-      <div>
-        {error}
-      </div>
+      <button onClick={() => setError("Error A")}>Throw Error A</button>
+      <button onClick={() => setError("Error B")}>Throw Error B</button>
+      <button onClick={() => setError("")}>Clear Errors</button>
+      <div>{error}</div>
     </div>
-  )
+  );
 }
 
 export function DangerousIdentifier() {
-  const [people, setPeople] = useState([])
+  const [people, setPeople] = useState([]);
 
   const handleSubmit = (e) => {
-    e.preventDefault()
-    const person = new FormData(e.target);
-    setPeople(ppl => [...ppl, ...person])
-  }
+    e.preventDefault();
+    const data = new FormData(e.currentTarget);
+    const person = data.get("person");
+    console.log(person);
+    setPeople((ppl) => [...ppl, { name: person }]);
+  };
 
-  return(
+  return (
     <div>
       <form onSubmit={handleSubmit}>
-        <input type="text" />
+        <input type="text" name="person" />
         <button>Add Person</button>
       </form>
       <ul>
-        {people.map(person => <span key={person.name}>{person.name}</span>)}
+        {people.map((person) => (
+          <span key={person.name}>{person.name}</span>
+        ))}
       </ul>
     </div>
-  )
+  );
 }
 
-async function fetchLeader() { return { name: 'Messi' }}
-async function fetchDetails(leader) { return { ...leader, country: 'Argentina' }}
+async function fetchLeader() {
+  return { name: "Messi" };
+}
+async function fetchDetails(leader) {
+  return { ...leader, country: "Argentina" };
```

Comment:
This is also generating an unnecessary copy of leader I think, it could just add country and mutate the original